src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestAttributeValueResolver.php

```diff
@@ -27,7 +27,27 @@ final class RequestAttributeValueResolver implements ArgumentValueResolverInterf
      */
     public function supports(Request $request, ArgumentMetadata $argument): bool
     {
-        return !$argument->isVariadic() && $request->attributes->has($argument->getName());
+        if ($argument->isVariadic()) {
+            return false;
+        }
+
+        if (!$request->attributes->has($argument->getName())) {
+            return false;
+        }
+
+        $type = $argument->getType();
+        // for union types or no type we assume it is supported to keep things simple
+        if (null === $type || str_contains($type, '|')) {
```

Comment:
```suggestion
        if (null === $type || $type instanceof \ReflectionUnionType) {
```

Reply:
The type is a string in ArgumentMetadata it already gets stringified by ArgumentMetadataFactory::getType

Reply:
Oh ok sorry